flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/FileSystemITCaseBase.scala

```diff
@@ -368,6 +372,63 @@ trait FileSystemITCaseBase {
     )
   }
 
+  @Test
+  def testReadAllMetadata(): Unit = {
+    if (!supportsReadingMetadata) {
+      return
+    }
+
+    tableEnv.executeSql(
+      s"""
+         |create table metadataTable (
+         |  x string,
+         |  filepath string metadata,
+         |  filename string metadata,
+         |  size bigint metadata,
+         |  modification_time timestamp_ltz metadata
+         |) with (
+         |  'connector' = 'filesystem',
+         |  'path' = '$resultPath',
+         |  ${formatProperties().mkString(",\n")}
+         |)
+         """.stripMargin
+    )
+
+    tableEnv.executeSql(
+      "insert into nonPartitionedTable (x) select x from originalT limit 1").await()
+
+    checkPredicate(
+      "select * from metadataTable",
+      row => {
+        assertEquals(5, row.getArity)
+
+        // Only one file, because we don't have partitions
+        val file = new File(URI.create(resultPath).getPath).listFiles()(0)
+
+        assertEquals(
+          file.getPath,
+          row.getFieldAs[String](1)
+        )
+        assertEquals(
+          Paths.get(file.toURI).getFileName.toString,
+          row.getFieldAs[String](2)
+        )
+        assertEquals(
+          file.length(),
+          row.getFieldAs[Long](3)
+        )
+        assertEquals(
+          // Note: It's TIMESTAMP_LTZ
+          Instant.ofEpochMilli(file.lastModified())
+            .atZone(DateTimeUtils.UTC_ZONE.toZoneId)
```

Comment:
why is `Instant.ofEpochMilli(file.lastModified())` not enough?

Reply:
Because `file.lastModified()` millis is adjusted to the local time zone. But the type here is TIMESTAMP_LTZ, which means that effectively when we write the record, the timestamp needs to be adjusted to UTC. That's why all this time weirdness (and that's the reason for below comment as well)

Reply:
Is this documented somewhere? Or can this be different per filesystem?

Because I see this in `FileStatus`:
```
    /**
     * Get the modification time of the file.
     *
     * @return the modification time of file in milliseconds since January 1, 1970 UTC.
     */
    long getModificationTime();
```

So UTC?

Reply:
I must be honest, I don't know if there's a difference here per filesystem. I'm pretty sure the local filesystem returns the time in the local time zone, because if you look how `FileStatus` is built, it invokes this method: https://docs.oracle.com/javase/8/docs/api/java/io/File.html#lastModified--